noir-projects/noir-protocol-circuits/crates/types/src/merkle_tree/variable_merkle_tree.nr

```diff
@@ -49,8 +50,8 @@ fn get_prev_power_2(value: u32) -> u32 {
         get_next_power_exponent(value, 0)
     };
 
-    let next_power_2 = 2 << next_power_exponent;
-    let prev_power_2 = next_power_2 / 2;
+    let prev_power_2 = 1 << (next_power_exponent - 1);
+    let next_power_2 = prev_power_2 * 2;
```

Comment:
I sneaked in an optimization. I compute prev_power_2 first and then I multiply by 2 to get next_power_2. This should be cheaper constraint-wise as division is more expensive than multiplication.